### DISCUSSION TITLE
Upgrading to WW 2.16 when we have github sites from which we do pull requests

### DISCUSSION THREAD

**Discussion by paulvojta** [openwebwork/webwork2#1483] [Q&A]
Dear all,
I sometimes do pull requests, and so have a github repository paulvojta/webwork2 for that purpose.
Webwork 2.16 has changed the name of the branch from "master" to "main", and the upgrade instructions include instructions for how to handle that ("git branch -m master main", etc.)
What do I need to do to change paulvojta/webwork2 to make it compatible with upstream?

**Comment by drgrice1** [comment]
The easiest thing to do would be to delete the master branch in your fork, and then push the main branch from the origin to your fork.

I guess I am assuming that you have a clone with multiple remotes.  One of which is openwebwork/webwork2 and one of which is your fork pauvojta/webwork2.  If you don't have this setup, I highly recommend it.  To create this setup do
```bash
git clone https://github.com/openwebwork/webwork2.git
cd webwork2
git remote add paulvojta https://github.com/paulvojta/webwork2.git
git fetch --all
```
Once that is done you will have two remotes named origin and paulvojta.  The first will have all of the branches from openwebwork/webwork2, and the second will have all of your branches (develop, master, release/2.7legacy, and release/2.8).  You can run `git branch -a` to see all of these.

If you already have a clone of openwebwork/webwork2 then you can skip the first step and just cd into that clone.  Then you will just be adding your fork as a remote to your existing clone.

Now you can run the following to get the main branch into your fork:
```bash
git push paulvojta main
```

If you don't have modifications to your master branch that you want to keep, then you can delete that with:
```
git push --delete paulvojta master
```
